utils/report.js

```diff
@@ -69,7 +68,29 @@ function handleExport(req, res, Model) {
   }
 }
 
+async function handleFindReturn(req, res, Model) {
+  try {
+    const parsedData = parseQueryData(req.body.query);
+    const query = buildQuery(parsedData);
+    var records = await Model.find(query,includeNgoFields).limit(exportLimit);
```

Comment:
Could you try to upgrade the handleSearch method to handle the projection and use the same for all filtering queries. Maybe you could use mongoose select to implement this.